lib/beaker-hostgenerator/hypervisor/abs.rb

```diff
@@ -24,7 +24,12 @@ def generate_node(node_info, base_config, bhg_version)
 
         case node_info['ostype']
         when /^(almalinux|centos|oracle|redhat|rocky|scientific)/
-          base_config['template'] ||= base_config['platform']&.gsub(/^el/, ::Regexp.last_match(1))
+          base_template = base_config['platform']&.gsub(/^el/, ::Regexp.last_match(1))
+          base_config['template'] ||= if node_info['bits'] == 'AARCH64' && node_info['ostype'] =~ /rocky|almalinux/
+                                        base_template&.gsub('aarch', 'arm')
+                                      else
+                                        base_template
+                                      end
```

Comment:
Note we already translate `aarch` to `arm` for redhat in the default case and when using `abs`
```
$ bundle exec beaker-hostgenerator redhat9-AARCH64a
---
HOSTS:
  redhat9-AARCH64-1:
    platform: el-9-aarch64
    template: redhat-9-arm64
    hypervisor: vmpooler
    roles:
    - agent
CONFIG:
  pooling_api: https://vmpooler-prod.k8s.infracore.puppet.net/
$ bundle exec beaker-hostgenerator redhat9-AARCH64a --hypervisor abs
---
HOSTS:
  redhat9-AARCH64-1:
    platform: el-9-aarch64
    template: redhat-9-arm64
    hypervisor: abs
    roles:
    - agent
CONFIG: {}
```

So I think you'll want to do something similar to how we're already handling this for RHEL 9: https://github.com/voxpupuli/beaker-hostgenerator/blob/62d51f1e17dba19350c061900f5a6017df48e824/lib/beaker-hostgenerator/data.rb#L387-L392

Also I'd expect some tests to be updated as a result of this change?

Reply:
I have updated the changes as you suggested. Please review.